File: models/FamilyMember.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const familyMemberSchema = new Schema({
    name: {
        type: String,
        required: true,
    },
    nationalId: {
        type: String,
        required: true
    },
    gender: {
        type: String,
        required: true
    },
    dateOfBirth: {
        type: String,
        required: true
    },
    relationship: {
        type: String,
        required: true
    }
}, { timestamps: true });

const FamilyMember = mongoose.model('FamilyMember', familyMemberSchema, "FamilyMember");
module.exports = FamilyMember;

File: clinic_front_end/lib/Models/FamilyMember.dart
class FamilyMember {
  String name = "";
  String nationalId = "";
  String gender = "";
  String dateOfBirth = "";
  String relation = "Son";

  FamilyMember(
      {required this.name,
      required this.nationalId,
      required this.gender,
      required this.dateOfBirth});

  FamilyMember.fromJson(Map<String, dynamic> json) {
    name = json['Name'];
    nationalId = json['NationalID'];
    gender = json['Gender'];
    dateOfBirth = json['DateOfBirth'];
    relation = json['Relationship'] ?? "Son";
  }
}
